packages/wrangler/src/dev.tsx

```diff
@@ -626,12 +626,17 @@ export async function startApiDev(args: StartDevOptions) {
 	};
 }
 /**
+ * Get an available TCP port number.
+ *
  * Avoiding calling `getPort()` multiple times by memoizing the first result.
  */
 function memoizeGetPort(defaultPort?: number) {
 	let portValue: number;
 	return async () => {
-		return portValue || (portValue = await getPort({ port: defaultPort }));
+		// Check a specific host to avoid probing all local addresses.
```

Comment:
There is one case where `memoizeGetPort()` needs to get a port on a potentially user specified IP address, which is the `getLocalPort()` case.
In this case we should already know what IP address is going to be used when we call this, so we could pass the IP address down so that it is used when checking whether a port is available, rather than just always using `localhost`.
Let me know if I am mistaken here.

Reply:
You are right! The `getLocalPort` value can indeed vary depending on the user-specified `--ip` option, or the platform-specific default. I will update it.

The other two ports are indeed localhost only, so I will just use the name. It would be unusual if an existing wrangler/workerd process is listening `127.0.0.1` and switching to `::1`. Or the other way round.